index.ts

```diff
@@ -822,6 +876,51 @@ class IntegrationAPI extends EventEmitter {
   public updateEntityAttributes(entityId: string, attributes: { [key: string]: string | number | boolean }): boolean {
     return this.#configuredEntities.updateEntityAttributes(entityId, attributes);
   }
+
+  public async generateOauth2AuthUrl(): Promise<{ auth_url: string }> {
+    const conn = this.#getCoreClient();
+    if (!conn) {
+      log.warn("generateOauth2AuthUrl: expected 1 client connected");
+      return Promise.reject("expected 1 client connected");
+    }
+    return this.#sendRequest(conn, generateRandomId(), api.MsgEvents.GenerateOauth2AuthUrl);
+  }
+
+  public async createOauth2Config(data: { [key: string]: any }) {
```

Comment:
The parameter seems to be mixed up with `generate_oauth2_auth_url`!
- `generate_oauth2_auth_url` allows an optional dictionary.
- `create_oauth2_cfg` requires `token_id`, `name`, `token` fields: https://github.com/unfoldedcircle/core-api/blob/77f3eed73a1c0e29e9aeacfa3026f5a71e1ebf21/integration-api/UCR-integration-asyncapi.yaml#L1522-L1547
  - These parameters need to be typed